pkg/scheduler/framework/runtime/framework.go

```diff
@@ -247,6 +256,11 @@ func NewFramework(r Registry, profile *config.KubeSchedulerProfile, stopCh <-cha
 		opt(&options)
 	}
 
+	logger := klog.TODO()
```

Comment:
Of course, it is also possible, but I think this PR itself is relatively large. If I change the `stopCh` parameter, more files will be involved. Maybe we can make changes in subsequent PR.

Reply:
> Instead you can pass a context at first, and then find out whether we can remove the stopCh later, but better to add a TODO here.

I looked at the calling path of this function `runtime.NewFramework`. 
```
scheduler.New -> profile.NewMap -> frameworkruntime.NewFramework
```
If the function `frameworkruntime.NewFramework` adds `context.Context ` parameter, the  `scheduler.New` function should preferably also have the context parameter. If this is the case, we can merge  PR  #116849 first, because  context parameter has already been added in PR #116849 

I'll look into this part of the code.

Reply:
> AFAIK we only use the WithLogger in tests because we can not pass the logger to framework right now. So we add a new field and a new wrapper, but actually, we can do this by pass-in a context.

We can use a logger that gets passed into the function in one of two ways:
- as explicit parameter
- attached to a context.Context

Obviously, the previous PR #111155 chose to pass the logger parameter in order not to change the `frameworkruntime.NewFramework` function parameters

However, `WithLogger` is not called in the following location, so no logger parameter is passed to the Framework.
https://github.com/kubernetes/kubernetes/blob/d060d487dcd74bc1a557f9378035696db40c64c2/pkg/scheduler/scheduler.go#L288


I think the correct code would look something like this:
```diff
diff --git a/pkg/scheduler/scheduler.go b/pkg/scheduler/scheduler.go
index 103c64c6dfa..26a49b9361c 100644
--- a/pkg/scheduler/scheduler.go
+++ b/pkg/scheduler/scheduler.go
@@ -246,6 +246,8 @@ func New(ctx context.Context,
        client clientset.Interface,
        stopCh <-chan struct{},
        opts ...Option) (*Scheduler, error) {
 
+       // Suppose https://github.com/kubernetes/kubernetes/pull/116849 has been merged
+       // we have the ctx variable
+       logger := klog.FromContext(ctx)
        stopEverything := stopCh
        if stopEverything == nil {
                stopEverything = wait.NeverStop
@@ -297,6 +299,7 @@ func New(client clientset.Interface,
                frameworkruntime.WithParallelism(int(options.parallelism)),
                frameworkruntime.WithExtenders(extenders),
                frameworkruntime.WithMetricsRecorder(metricsRecorder),
+               frameworkruntime.WithLogger(logger),
        )
        if err != nil {
                return nil, fmt.Errorf("initializing profiles: %v", err)

```

So, based on the above investigation, for the `pkg/scheduler/framework/runtime/` package, we can now choose the following two ways to get a logger:
- Pass explicit parameter logger, use `frameworkruntime.WithLogger` function to pass
- Pass the `context.Context` parameter, we can also remove the `stopCh` parameter afterwards/immediately

I don't know much about the code of scheduler, please help to confirm it.
/cc @pohly  @kerthcet @alculquicondor

Reply:
So what's the long term plan about the `logger := klog.TODO()`? I think we need to remove it one day. Pass a context to `NewFramework` will touch a lot of codes, let's come to a consequence at first. cc @alculquicondor Any advices?

Reply:
> We can use a logger that gets passed into the function in one of two ways:
   as explicit parameter
   attached to a context.Context

Seems we already follow the approach 2 when building scheduler in https://github.com/kubernetes/kubernetes/pull/111155/files#diff-7a5cd3adb38f371f579372a2c901e73675f590515b35c4a8344217a91eff11fdR338-R340. Why not pass the context to framework as well. No another logger option.

Reply:
> Why not pass the context to framework as well. No another logger option.

What was the reason for the initial decision to use the logger option? @pohly

Reply:
Passing a `logger` is usually the safe choice (a bit less overhead, no additional semantic added to the function call beyond logging). If there are additional reasons for passing a context (for example, handling cancellation), then a context is better.

For `NewFramework`, there is a `stopCh`. That can be replaced with a `context.Done()`, so here passing a `ctx` is better.

Reply:
@mengjiao-liu can you acknowledge that the correct solution is to pass the context and that you will open a follow up PR to fix this?

Reply:
Ok, I will open a PR to follow up immediately and pass the context.